Ginger/GingerCoreNET/AnalyzerLib/AnalyzeAction.cs

```diff
@@ -240,55 +240,63 @@ public static List<AnalyzerItemBase> Analyze(BusinessFlow BusinessFlow, Activity
+                                    if (WorkSpace.Instance.Solution.Variables.Where(x => x.SupportSetValue && x.Guid.ToString() == ARV.StoreToValue).FirstOrDefault() == null)
                                     {
-                                        AnalyzeAction AA = CreateNewIssue(BusinessFlow, parentActivity, a);
-                                        AA.Description = chkDataSource;
-                                        AA.Details = "Invalid '" + StoreToValue + "' StoreTo Value used In " + GingerDicser.GetTermResValue(eTermResKey.BusinessFlow) + " '" + BusinessFlow.Name + "' => " + GingerDicser.GetTermResValue(eTermResKey.Activity) + " '" + parentActivity.ActivityName + "' =>" + "Action '" + a.Description + "' ";
-                                        AA.HowToFix = "Remap Output Values";
-                                        AA.CanAutoFix = AnalyzerItemBase.eCanFix.No;
-                                        AA.IssueType = eType.Error;
-                                        AA.Impact = "Execution will fail in run time";
-                                        AA.Severity = eSeverity.Critical;
-                                        IssuesList.Add(AA);
+                                        issueFound = true;
+                                    }
+                                    break;
 
+                                case ActReturnValue.eStoreTo.ApplicationModelParameter:
+                                    if (WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<GlobalAppModelParameter>().Where(x => x.Guid.ToString() == ARV.StoreToValue).FirstOrDefault() == null)
+                                    {
+                                        issueFound = true;
                                     }
-                                }
+                                    break;
+
+                                case ActReturnValue.eStoreTo.DataSource:
+                                    Regex rxDSPattern = new Regex(@"{(\bDS Name=)\w+\b[^{}]*}", RegexOptions.Compiled);
+                                    MatchCollection matches = rxDSPattern.Matches(ARV.StoreToValue);
+                                    foreach (Match match in matches)
+                                    {
+                                        if (GingerCoreNET.GeneralLib.General.CheckDataSource(match.Value, DSList) != "")
+                                        {
+                                            issueFound = true;
+                                        }
+                                    }
+                                    break;
                             }
                         }
+                        if (issueFound)
+                        {
+                            AnalyzeAction AA = CreateNewIssue(BusinessFlow, parentActivity, a);
+                            AA.Description = string.Format("Output Value StoreTo has in-valid configuration");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this formatting call and simply use the input string.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5538205)